src/monodroid/jni/monodroid-glue.c

```diff
@@ -535,6 +536,28 @@ get_xamarin_android_msbuild_path (void)
 	return msbuild_folder_path;
 }
 
+static char *libmonoandroid_directory_path = NULL;
+
+// Returns the directory in which this library was loaded from
+static char*
+get_libmonoandroid_directory_path ()
+{
+	wchar_t module_path[MAX_PATH];
+	char *converted_path = NULL;
+	HMODULE module = NULL;
+
+	if (libmonoandroid_directory_path != NULL)
+		return libmonoandroid_directory_path;
+
+	if (!GetModuleHandleEx (GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
```

Comment:
I don't think that this has explicitly come up before, but *in general*, we should follow the precepts in [UTF-8 Everywhere](http://utf8everywhere.org), *especially* the [Windows guidelines](http://utf8everywhere.org/#windows):

* Do not use types, functions, or their derivatives that are sensitive to the `UNICODE` constant, such as `LPTSTR`, `CreateWindow()` or the `_T()` macro. Instead, use `LPWSTR`, `CreateWindowW()` and explicit `L""` literals.
* Only use Win32 functions that accept widechars (`LPWSTR`), never those which accept `LPTSTR` or `LPSTR`.

Consequently, this should use `GetModuleHandleExW()` explicitly, *not* `GetModuleHandleEx()`. Passing `& libmonoandroid_directory_path` is also *really* suspect to my eyes; it may work, but...ugh.

Instead, I'd want to declare at global scope:

    static WCHAR for_address_computation;

Then we don't need a cast:

```c
if (!GetModuleHandleExW (
            GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
            &for_address_computation,
            &module))
    return NULL;
```